File: src/features/payments/ui/receiptScreenData.ts
export type ReceiptScreenData =
  | { kind: 'notFound' }
  | { kind: 'processing'; billerName: string; amountPaise: number }
  | {
      kind: 'success';
      paymentId: string;
      billerId: string;
      billerName: string;
      account: string;
      amountPaise: number;
      paidAt: string;
      canSaveBiller: boolean;
    }
  | {
      kind: 'failed';
      billerId: string;
      billerName: string;
      categoryId: string;
      account: string;
      amountPaise: number;
    };

File: lib/features/payments/ui/receipt_screen_data.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'receipt_screen_data.freezed.dart';

@freezed
sealed class ReceiptScreenData with _$ReceiptScreenData {
  const factory ReceiptScreenData.notFound() = ReceiptNotFound;
  const factory ReceiptScreenData.processing({
    required String billerName,
    required int amountPaise,
  }) = ReceiptProcessing;
  const factory ReceiptScreenData.success({
    required String paymentId,
    required String billerId,
    required String billerName,
    required String account,
    required int amountPaise,
    required DateTime paidAt,
    required bool canSaveBiller,
  }) = ReceiptSuccess;
  const factory ReceiptScreenData.failed({
    required String billerId,
    required String billerName,
    required String categoryId,
    required String account,
    required int amountPaise,
  }) = ReceiptFailed;
}
